### ISSUE TITLE
Download slows down after 10 seconds when downloading to USB on android TV

### ISSUE THREAD

**Issue by thim0o** [scakemyer/plugin.video.quasar#862]
When i try to download a torrent to a usb drive attached to my sony android tv (kodi 17.3) it first starts downloading at the speeds it should be downloading (about 8000kb/s) but after about 10 seconds it quickly decreases to 40-50 kb/s.
Is anyone else having this issue aswell? The version is v0.9.78. Also i tested to download to the usb from a windows computer and that's working as it should. So no issue with the usb drive. 

A torrent download to the internal storage of the tv worked fine a few months ago.

Things that i've tried:

Rebooted the TV
Fresh install
Reformat the usb to NTSF and exFAT
Limit max connections
Limit download and upload speed
Tried old versions
Installed in SPMC 16
Nothing seems to fix this problem

I don't have a log file because it's not that easy to get on android tv, but if it's really needed i can post one.

Any help will be greatly appreciated

**Comment by mrwsl** [comment]
I had this issue as well. Im using a Intel NUC and what solved it for me was that I switched the usb port for the external drive.

**Comment by thim0o** [comment]
```
I had this issue as well. Im using a Intel NUC and what solved it for me was that I switched the usb port for the external drive.
```
The TV has one usb 1.0 one 2.0 and one 3.0. I tried it in usb 2 and 3. Both have exactly the same problem. Usb 1 isn't worth trying, right? Because of the 12 Mbit/s speed limitation.

**Comment by mrwsl** [comment]
I mean if the other two ports don't give you a better result it's worth trying the third, isn't it?

**Comment by thim0o** [comment]
Yes i Will try. But I think 12mbit/ 1,5mb on the usb1 per second is too slow. Not sure if it is USB 1 though. Will try.

**Comment by josete2k** [comment]
It's a libtorrent issue...

**Comment by thim0o** [comment]
> It's a libtorrent issue... 

Any fix or temporary workaround?

**Comment by thim0o** [comment]
No success with the other USB port. Also if I open the exact same torrent with "Yet another torrent player" for Kodi, it streams perfectly fine. So the problem is in the quasar add-on.

**Comment by josete2k** [comment]
https://github.com/scakemyer/plugin.video.quasar/issues/90

**Comment by thim0o** [comment]
It doesn't crash though. It just slows down to 40-50kbps and keeps downloading at that speed. So this is not the same problem but definitely related.

**Comment by josete2k** [comment]
Yes, it's the same problem.

Sometimes it slows down and ups in a few seconds... and you can see the stream, but many times it remains at 10kbs-5kbs-30kbs...

**Comment by iamacarpet** [comment]
Please check out #843 

We've got a build with gotorrent instead of libtorrent thanks to @elgatito, plus a FAT32 driver that's specially tuned to Android, which solves the slow down problem in most cases. Although, testing indicates a LOT better performance with MicroSD cards compared with cheap USB drives (premium ones perform a little better, which suggests cheap flash is the problem, not the fact they are USB).

**Comment by thim0o** [comment]
No i'm sure the issue is not with the usb drive. When i plug it in to my pc i can steam something in quasar to the usb perfectly fine.

**Comment by thim0o** [comment]
The version with gotorrent doesn't work at all for me.

**Comment by elgatito** [comment]
Getting error or what does not work?

**Comment by thim0o** [comment]
If I remember it correctly it just keeps loading when i try to open the addon. I'm not sure though. I will check it tomorrow to make sure.

**Comment by elgatito** [comment]
@thim0o , take the latest version here - https://github.com/elgatito/plugin.video.quasar/releases/tag/v0.9.78 and give a kodi.log with debug mode enabled.

**Comment by thim0o** [comment]
Update: I remember now, it does work but has the exact same problem with usb. However I'm not sure if I installed it correctly since I was in a bit of a hurry. Thanks for the link, I'll test it again tomorrow.

**Comment by elgatito** [comment]
You can also try the memory storage type (first tab of Quasar settings). Not finished, but you can look how it works for the moment.

**Comment by foklorr** [comment]
Newer USB flash [UFD] have a cache on first write.  Perhaps you are filling that and then hitting the WriteOut bottleneck.

**Comment by foklorr** [comment]
try both of these

https://amazon.com/gp/aw/d/B013UHK0TK/

https://amazon.com/gp/aw/d/B00OAJ412U/

**Comment by ikerfp** [comment]
Hi, where can I download elementum? because there are not releases at this moment https://github.com/elgatito/plugin.video.elementum/releases

**Comment by elgatito** [comment]
Sorry, no estimates yet. It will appear there when working version is ready. But thag will be a separate plugin.

**Comment by talrhv** [comment]
I've got the exact same problem with my Sony android tv and also with xiaomi mi box. Is there any solution?

**Comment by avshi114** [comment]
Can i ask how did you set download path to external drive?
I have a sony android tv device and all i see in the download path is the /sdcard path which is my internal memory and I cannot go up from this folder.

**Comment by joaopauloss** [comment]
Same problem here. One thing i tried to do is:

-Format your USB stick as NTFS format.
-Create a folder structure like this on the USB stick: \Android\data\org.xbmc.kodi\files\Quasar
-Insert USB stick.
-Open Quasar settings.
-Click on “Download path (not FAT32)” and select the folder we previously created: \Android\data\org.xbmc.kodi\files\Quasar
-Save the settings.

It  works with some drop of download speed (hardware limitation, of course), but nothing to compromise. 
E.g.: 2000 KB/s then a little drop to 1600 but go to 2000 again.

I have 3 USB sticks with 4, 8 and 16gb. Only the ones with 4 and 16 worked.

**Comment by frevib** [comment]
Formatting my Samsung evo plus 64gb microsd to NTFS solved the problem, @joaopauloss. I still see some drops from 8000KB/s to 2000KB/s but it goes back to 8000KB/s quickly. These minor drops could as well be due to bandwith availabilty in the swarm.

Running Kodi/Quasar on quad-core ARM 64 hardware with 1gb RAM. 

Edit: I did an extra check, no download speed drops so this issue is solved for me. The microsd card was originally formatted exfat.

**Comment by iamacarpet** [comment]
@joaopauloss Great instructions.

I'd like to add that the GoTorrent version is fairly mature now as [Elementum](https://github.com/elgatito/elementum).

There is a FAT32 driver specifically made for the old AFTV2 with the MicroSD slot where NTFS/exFAT isn't supported, plus @elgatito 's in memory driver is getting fairly mature and is very usable, so you don't need backing storage at all.

**Comment by bobnett** [comment]
Thanks, @joaopauloss  ... great instructions ... worked perfectly here on my tv box with a micro usb 64gb sandisk ...

**Comment by mixalbl4-127** [comment]
Thanks, @joaopauloss !!!
worked perfectly on my tv box with a micro usb 64gb Samsung
I searched 2 days and solved, thanks
P.S. The microsd card was originally formatted exfat.
